content/post/kubestash-v2024.12.9/index.md

```diff
@@ -0,0 +1,195 @@
+
+### New Features
+
+Here, we are going to highlight the new features that have been introduced in this release.
+
+#### KubeDB Managed Database Backup Verification
+
+We've introduced functionality to verify KubeDB managed database backup. You can configure following types of backup verification strategies:
+
+- **RestoreOnly :** KubeStash operator will initiate a `RestoreSession` using the addon information specified in `BackupVerifier`. The verification of the backup will rely on the status of the `RestoreSession` phase; if the restore completes successfully, the backup is considered verified.
+- **Query :** At first, KubeStash operator will initiate a `RestoreSession` and after successful restore, it will create a verifier job to run the queries provided in `BackupVerifier`.
+- **Script :** At first, KubeStash operator will initiate a `RestoreSession` and after successful restore, it will create a verifier job to run the script provided in `BackupVerifier`.
+
+Here we will give an example of query verification strategy for KubeDB managed `MySQL`.
+
+At first, we need to check if the verification function is present in our cluster using the following command:
+
+```bash
+$ kubectl get functions | grep -i `kubedbverifier`
+kubedbverifier            19h
+```
+
+Assuming that a KubeDB managed `MySQL` instance with a configured backup already exists. Now, we need to create a `MySQL` instance for verification in our preferred namespace (i.e. `verify`). Here is an example of `MySQL` instance for verification:
+
+```yaml
+apiVersion: kubedb.com/v1
+kind: MySQL
+metadata:
+  name: sample-mysql
+  namespace: verify
+spec:
+  version: "8.0.35"
+  replicas: 1
+  storageType: Durable
+  storage:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 50Mi
+```
+
+Here, the configuration for the verification application should be similar to the backup application as we are going to restore the backup data in the verification application.
+
+Here is the example of `BackupVerifier` that verifies the `MySQL`:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupVerifier
+metadata:
+  name: mysql-query-verifier
+  namespace: demo
+spec:
+  function: kubedbverifier
+  restoreOption:
+    target:
+      apiGroup: kubedb.com
+      kind: MySQL
+      name: sample-mysql
+      namespace: verify
+    addonInfo:
+      name: mysql-addon
+      tasks:
+        - name: logical-backup-restore
+  scheduler:
+    schedule: "*/5 * * * *"
+    jobTemplate:
+      backoffLimit: 1
+  type: Query
+  query:
+    mySQL:
+      - database: auth
+        table: user
+        rowCount:
+          operator: Equal
+          value: 100
+      - database: store
+        table: order
+  sessionHistoryLimit: 2
+```
+
+Now we need to update `BackupConfiguration` to refer the `BackupVerifier`. 
```

Comment:
Here is the BackupConfiguration configured for the original database, which refers to the BackupVerifier